sockeye/inference.py

```diff
@@ -952,20 +982,19 @@ def _get_inference_input(self, trans_inputs: List[TranslatorInput]) -> Tuple[mx.
         """
         bucket_key = data_io.get_bucket(max(len(inp.tokens) for inp in trans_inputs), self.buckets_source)
 
-        utils.check_condition(C.PAD_ID == 0, "pad id should be 0")
```

Comment:
This could be an assert.

Reply:
I moved it to the constructor. We anyway require this throughout the project so i wanted to remove the assert being called all the time.

Reply:
Ah! Hadn't seen it. Nevertheless this should be an assert, as it is a programming error and not a user error.